lib/charms/tls_certificates_interface/v3/tls_certificates.py

```diff
@@ -1767,10 +1762,9 @@ def _on_relation_changed(self, event: RelationChangedEvent) -> None:
         for certificate in provider_certificates:
             if certificate.csr in requirer_csrs:
                 if certificate.revoked:
-                    if JujuVersion.from_environ().has_secrets:
-                        with suppress(SecretNotFoundError):
-                            secret = self.model.get_secret(label=f"{LIBID}-{certificate.csr}")
-                            secret.remove_all_revisions()
+                    with suppress(SecretNotFoundError):
+                        secret = self.model.get_secret(label=f"{LIBID}-{certificate.csr}")
+                        secret.remove_all_revisions()
```

Comment:
I am not sure it makes sense to drop Juju 2 support at this time, the current LTS is still 2.9.

Reply:
I'm not suggesting to completely drop Juju 2 support. We can still maintain the tls lib v2. But the lib v3 won't support Juju 2.

Reply:
Is it now possible to push updates to older major versions of library? It was problematic in the past, and last I remember it was being worked on.

If we can, then I am fine with this.

Reply:
Yes, this was fixed

Ref: https://bugs.launchpad.net/snapstore-server/+bug/2024370

Reply:
Awesome!